### PR TITLE
VEN-1271 | Create boat for existing applications

### PR DESCRIPTION
**Pull request by a-rmz**
## Description :sparkles:
* Add a command that creates (or updates) boats for existing applications **with leases**
  * If the application has a lease, it assigns the boat to the lease
* Add validation to the boat fields on `BaseApplication` 
* Extract boat creation to separate util

### Closes :no_good_woman:
**[VEN-1271](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1271):** Create boats in database for previously handled applications

### Automated tests :gear:️
```shell
$ pytest applications/management/commands/create_boat_for_existing_applications.py
```

### Manual testing :construction_worker_man:
1. Create an application
2. Create a lease for that application
3. Move the application to `OFFERED` and the lease to `PAID`
4. Run the command `python manage.py create_boat_for_existing_applications`
5. Check the boats for the customer who submitted the application, there should be a new boat with the fields from the application

### CODE REVIEW

File: applications/management/commands/create_boat_for_existing_applications.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from django.core.management import BaseCommand
+
+from applications.models import BerthApplication, WinterStorageApplication
+from applications.utils import create_or_update_boat_for_application
+from leases.consts import ACTIVE_LEASE_STATUSES
+
+
+class Command(BaseCommand):
+    help = ""
+    boat_types = []
```
Comment (klemmari1) [inline applications/management/commands/create_boat_for_existing_applications.py:13]: Unused variable

File: applications/management/commands/create_boat_for_existing_applications.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+from django.core.exceptions import ValidationError
+from django.core.management import BaseCommand
+
+from applications.models import BerthApplication, WinterStorageApplication
+from applications.utils import create_or_update_boat_for_application
+from leases.consts import ACTIVE_LEASE_STATUSES
+
+
+class Command(BaseCommand):
+    help = ""
```
Comment (klemmari1) [inline applications/management/commands/create_boat_for_existing_applications.py:12]: Empty help text

File: applications/utils.py [hunk 3]
```diff
@@ -231,3 +233,23 @@ def localize_datetime(dt, language=settings.LANGUAGES[0][0]):
...
+        owner=application.customer,
+        registration_number=application.boat_registration_number,
+        defaults={
+            "boat_type_id": application.boat_type_id,
+            "length": application.boat_length,
+            "width": application.boat_width,
+            "registration_number": getattr(application, "boat_registration_number", ""),
+            "name": getattr(application, "boat_name", ""),
+            "model": getattr(application, "boat_model", ""),
+            "draught": getattr(application, "boat_draught", None),
```
[the remark this answers is not in the document]
Comment (a-rmz) [reply applications/utils.py:249]: Hmm good points, let's sync with the frontend to add the restrictions/validations from the customer UI.

For this case, if the draught is 0, I'll just make it `None` instead.

File: applications/management/commands/create_boat_for_existing_applications.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        if application.lease:
+            application.lease.boat = boat
+            application.lease.save()
+
+        if _created:
+            self.created += 1
+        else:
+            self.updated += 1
+
+    def handle(self, *args, **options):
```
[the remark this answers is not in the document]
Comment (a-rmz) [reply applications/management/commands/create_boat_for_existing_applications.py:34]: Done 👍

File: applications/management/commands/create_boat_for_existing_applications.py (hunk 2 above)
Comment (a-rmz) [reply applications/management/commands/create_boat_for_existing_applications.py]: Done 👍

File: applications/management/commands/create_boat_for_existing_applications.py (hunk 1 above)
Comment (a-rmz) [reply applications/management/commands/create_boat_for_existing_applications.py]: Done 👍

**Review by klemmari1 — APPROVED** [review]
Looks good to me!

**Pull request merged**